src/Button/Button.js

```diff
@@ -0,0 +1,102 @@
+import React, {Component, PropTypes} from 'react';
+import propTypes from './buttonPropTypes';
+import EnhancedButton from '../internal/EnhancedButton';
+import getStyles from './buttonStyles';
+import styleButtonChildren from './styleButtonChildren';
+
+import {
+  handleMouseDown,
+  handleMouseUp,
+  handleMouseLeave,
+  handleMouseEnter,
+  handleTouchStart,
+  handleTouchEnd,
+  handleKeyboardFocus,
+} from './buttonEventHandlers';
+
+class Button extends Component {
+  static muiName = 'Button';
+  static propTypes = propTypes;
+
+  static defaultProps = {
+    disabled: false,
+    fullWidth: false,
+    onKeyboardFocus: () => {},
+    onMouseDown: () => {},
+    onMouseEnter: () => {},
+    onMouseLeave: () => {},
+    onMouseUp: () => {},
+    onTouchEnd: () => {},
+    onTouchStart: () => {},
+    primary: false,
+    secondary: false,
+    type: 'raised',
```

Comment:
It's a fair question - I was minimising the prop count, but kept the current API for primary / secondary.

Having a boolean prop for each button type would let users do `<Button flat />` if they don't enforce `={true}`. 

OTOH, I could change primary/secondary to a single string prop...